pkg/ccl/backupccl/restore_planning.go

```diff
@@ -802,6 +802,45 @@ func resolveTargetDB(
 	return database.Name, nil
 }
 
+// maybeUpgradeTableDescsInSlice updates the passed slice of table descriptors
+// to use the newer 19.2-style foreign key representation, if they are not
+// already upgraded. This requires resolving cross-table FK references, which is
+// done by looking up all table descriptors in the slice provided.
+//
+// if skipFKsWithNoMatchingTable is set, FKs whose "other" table is missing from
+// the set provided are omitted during the upgrade, instead of causing an error
+// to be returned.
+func maybeUpgradeTableDescsInSlice(
+	ctx context.Context, descs []catalog.Descriptor, skipFKsWithNoMatchingTable bool,
+) error {
+	descGetter := catalog.MapDescGetter{}
+
+	// Populate the protoGetter with all table descriptors in all backup
+	// descriptors so that they can be looked up.
+	for _, desc := range descs {
+		desc.DescriptorProto().GetTable()
+		if table := desc.DescriptorProto().GetTable(); table != nil {
+			descGetter[table.ID] = desc
```

Comment:
Just to stay on the safe side you might want to consider deep-copying this value: `*protoutil.Clone(desc).(*descpb.Descriptor)`

Reply:
I'm a bit lost in the wrapping/unwrapping/conversions to/from proto, since desc isn't a proto

Reply:
Never mind, sorry, this isn't how deep-copying works. Sorry for the confusion. In any case, I checked further, and it's actually safe to not deep-copy here.